core/src/tasks/profile/completeImport.ts

```diff
@@ -16,57 +16,62 @@ export class ProfileCompleteImport extends RetryableTask {
     this.frequency = 0;
     this.queue = "profiles";
     this.inputs = {
-      profileId: { required: true },
+      profileIds: { required: true },
       toExport: { required: true },
     };
   }
 
   async runWithinTransaction(params) {
-    const profile = await Profile.findOne({
-      where: { id: params.profileId },
-      include: [ProfileProperty],
+    const properties = await Property.findAllWithCache();
+    const profiles = await Profile.findAll({
+      where: { id: { [Op.in]: params.profileIds } },
+      include: [
+        { model: ProfileProperty, required: true },
+        { model: Import, required: false, where: { profileUpdatedAt: null } },
+      ],
     });
-    if (!profile) return; // the profile may have been deleted or merged by the time this task ran
+    if (profiles.length === 0) return;
 
-    const properties = await Property.findAllWithCache();
-    const pendingProfileProperty = profile.profileProperties.find(
-      (p) => p.state !== "ready" // a property may have gone back into the pending state
+    const readyProfiles = profiles.filter(
+      (profile) =>
+        profile.state === "ready" &&
+        !profile.profileProperties.find((p) => p.state !== "ready")
     );
+    if (readyProfiles.length === 0) return;
 
-    if (profile.state !== "ready" || pendingProfileProperty) {
-      return CLS.enqueueTaskIn(config.tasks.timeout + 1, this.name, params);
-    }
```

Comment:
This re-enqueue was likely the problem.  If we keep re-enqueueing n+1 tasks (where n > than the number of workers) we'll never complete this queue, and won't move onto anything else.   The `profiles` queue is higher in priority than `profileProperties`... so we won't import anything.

Now, we'll just wait until the next `checkReady` batch to try again.  It's safer although a little slower... but the case where a profile is made un-ready by a new property or import in the middle of another import is rare.